domain/src/main/kotlin/no/nav/su/se/bakover/domain/kontrollsamtale/Kontrollsamtale.kt

```diff
@@ -0,0 +1,63 @@
+package no.nav.su.se.bakover.domain.kontrollsamtale
+
+import no.nav.su.se.bakover.common.Tidspunkt
+import no.nav.su.se.bakover.common.endOfMonth
+import no.nav.su.se.bakover.common.periode.Periode
+import no.nav.su.se.bakover.common.startOfMonth
+import no.nav.su.se.bakover.common.zoneIdOslo
+import no.nav.su.se.bakover.domain.vedtak.Vedtak
+import java.time.Clock
+import java.time.LocalDate
+import java.util.UUID
+
+data class Kontrollsamtale(
+    val id: UUID = UUID.randomUUID(),
+    val opprettet: Tidspunkt = Tidspunkt.now(),
+    val sakId: UUID,
+    val innkallingsdato: LocalDate?,
+    val status: Kontrollsamtalestatus,
+    val frist: LocalDate?
+) {
+    companion object {
+        fun fromVedtak(vedtak: Vedtak): Kontrollsamtale {
+            val innkallingsdato = regnUtInnkallingsdato(vedtak.periode, vedtak.opprettet.toLocalDate(zoneIdOslo))
+
+            return Kontrollsamtale(
+                sakId = vedtak.behandling.sakId,
+                innkallingsdato = innkallingsdato,
+                status = Kontrollsamtalestatus.PLANLAGT_INNKALLING,
+                frist = innkallingsdato?.endOfMonth()
+            )
+        }
+    }
+}
+
+enum class Kontrollsamtalestatus(val value: String) {
+    PLANLAGT_INNKALLING("PLANLAGT_INNKALLING"),
+    INNKALT("INNKALT"),
+    GJENNOMFØRT("GJENNOMFØRT")
+}
+
+private fun LocalDate.isLessThanOneMonthAfter(localDate: LocalDate) = this.isBefore(localDate.plusMonths(1))
+
+fun regnUtInnkallingsdato(periode: Periode, vedtaksdato: LocalDate, clock: Clock? = null): LocalDate? {
+    val stønadsstart = periode.fraOgMed
+    val stønadsslutt = periode.tilOgMed
+    val fourMonthsDate = stønadsstart.plusMonths(4).startOfMonth()
+    val eightMonthsDate = fourMonthsDate.plusMonths(4)
+    val today = if (clock == null) LocalDate.now() else LocalDate.now(clock)
+
+    return if (eightMonthsDate.isAfter(today) && !fourMonthsDate.isAfter(today)) {
+        when {
+            stønadsslutt.isLessThanOneMonthAfter(eightMonthsDate) -> null
+            eightMonthsDate.isLessThanOneMonthAfter(vedtaksdato.endOfMonth()) -> eightMonthsDate.plusMonths(1)
```

Comment:
eightMonthsDate.plusMonths(1) kan vel i teorien være nærmere stønadsslutt enn 1 måned? (samme for 4. mnd)

Reply:
Godt observert!